nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -105,33 +110,63 @@ boolean isConnected() {
         return socket != null && socket.isConnected();
     }
 
+    private InetSocketAddress inetSocketAddress() {
+        DnsResolver dnsResolver = connectionKey.dnsResolver();
+        if (dnsResolver.useDefaultJavaResolver()) {
+            return new InetSocketAddress(connectionKey.host(), connectionKey.port());
+        } else {
+            InetAddress address = dnsResolver.resolveAddress(connectionKey.host(), connectionKey.dnsAddressLookup());
+            return new InetSocketAddress(address, connectionKey.port());
+        }
+    }
+
+    private int connectToProxy(InetSocketAddress remoteAddress) throws IOException {
+        String hostPort = remoteAddress.getHostName() + ":" + remoteAddress.getPort();
+        // FIXME Ideally we would pass the socket of this connection into the next Http1Client
+//        Http1Client proxyConnection = Http1Client.builder()
+//                .baseUri("http://" + hostPort).build();
+//        Http1ClientResponse response = proxyConnection.method(Method.CONNECT)
+//                .header(Header.HOST, hostPort).request();
+//        return response.status().code();
+
+        // FIXME This has to be replaced by previous lines
+        String newLine = "\r\n";
+        HelidonSocket tempSocket = PlainSocket.client(socket, createChannelId(socket));
+        StringBuilder httpConnect = new StringBuilder();
+        httpConnect.append("CONNECT ").append(hostPort).append(" HTTP/1.1").append(newLine)
+        .append("Host: ").append(hostPort).append(newLine)
+        .append("Accept: */*").append(newLine).append(newLine);
+        // TODO Authentication
```

Comment:
Follow-up issue and remove TODO

Reply:
Done, I will create the issue once this is merged.